cpp/src/lib/proxy_protocol.cc

```diff
@@ -391,32 +394,74 @@ void read_ping_response(message &m, Status &status, double &timestamp) {
 
 void write_osd_info_request(message_builder &mb) { write_tag(mb, _OSD_INFO); }
 
-void read_osd_info_response(
-    message &m, Status &status,
-    std::vector<std::pair<osd_t, std::pair<std::unique_ptr<OsdInfo>,
-                                           std::unique_ptr<OsdCapabilities>>>> &
-        result) {
+void _read_osd_infos(message &m, osd_map_t &result) {
+  uint32_t n;
+  from(m, n);
+  for (uint32_t i = 0; i < n; i++) {
+    osd_t osd_id;
+    from(m, osd_id);
+    std::string info_s;
```

Comment:
hadn't really given this any thought, it was already like this when I found it ;-), but I'll try to make an improvement here